seaborn/categorical.py

```diff
@@ -1797,78 +1799,72 @@ def __init__(self, x, y, hue, data, order, hue_order,
                  width, dodge, k_depth, linewidth, scale, outlier_prop,
                  showfliers=True):
 
-        # TODO assigning variables for None is unneccesary
-        if width is None:
-            width = .8
         self.width = width
-
         self.dodge = dodge
-
-        if saturation is None:
-            saturation = .75
         self.saturation = saturation
 
-        if k_depth is None:
-            k_depth = 'proportion'
+        if k_depth not in ('proportion', 'tukey', 'trustworthy') and (
+            not isinstance(k_depth, numbers.Number)
+        ):
+            raise ValueError(f'k_depth "{k_depth}" not recognized')
         self.k_depth = k_depth
 
+        # TODO seems this member is only used to frame the legend artists
         if linewidth is None:
             linewidth = mpl.rcParams["lines.linewidth"]
         self.linewidth = linewidth
 
-        if scale is None:
-            scale = 'exponential'
+        if scale not in ('linear', 'exponential', 'area'):
+            raise ValueError(f'Scale "{scale}" not recognized')
         self.scale = scale
 
+        if ((outlier_prop > 1) or (outlier_prop < 0)):
+            msg = f'outlier_prop {outlier_prop} not in range [0, 1]'
+            raise ValueError(msg)
         self.outlier_prop = outlier_prop
 
         self.showfliers = showfliers
 
         self.establish_variables(x, y, hue, data, orient, order, hue_order)
         self.establish_colors(color, palette, saturation)
 
-    def _lv_box_ends(self, vals, k_depth='proportion', outlier_prop=None):
+    def _lv_box_ends(self, vals):
         """Get the number of data points and calculate `depth` of
         letter-value plot."""
         vals = np.asarray(vals)
         vals = vals[np.isfinite(vals)]
         n = len(vals)
-        # If p is not set, calculate it so that 8 points are outliers
-        if not outlier_prop:
-            # Conventional boxplots assume this proportion of the data are
-            # outliers.
-            p = 0.007
-        else:
-            if ((outlier_prop > 1.) or (outlier_prop < 0.)):
-                raise ValueError('outlier_prop not in range [0, 1]!')
-            p = outlier_prop
+        p = self.outlier_prop
+
         # Select the depth, i.e. number of boxes to draw, based on the method
-        k_dict = {
-            'proportion': (np.log2(n)) - int(np.log2(n * p)) + 1,
-            'tukey': (np.log2(n)) - 3,
-            'trustworthy': 1 + (np.log2(n)
-                                - np.log2(2 * stats.norm.ppf((1 - p)) ** 2))
-        }
-        k = k_dict[k_depth]
-        try:
-            k = int(k)
-        except ValueError:
-            k = 1
-        # If the number happens to be less than 0, set k to 0
-        if k < 1.:
+        if p == 0:
```

Comment:
Do I understand correctly that `outlier_prop` is only relevant when `k_depth == 'proportion`' *or* if it is set to 0? That might be ok, but it does seem a little convoluted.

Maybe `k_depth="full"` would be cleaner?

Reply:
Indeed. You suggestion is much better, will add "full". Note that this will not imply last box goes all the way towards the outliers; just having log(n)+1 boxes. It will be clearly documented.